File: dxtrade-extension-backend/src/main/java/org/chrome/extension/dto/request/PlaceOrderRequest.java
package org.chrome.extension.dto.request;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.chrome.extension.domain.Broker;
import org.chrome.extension.domain.OrderType;
import org.chrome.extension.domain.PositionSide;

import java.math.BigDecimal;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class PlaceOrderRequest {
    private String broker;
    private String apiKey;
    private String orderId;
    private String accountCode;
    private BigDecimal quantity;
    private String instrument;
    private OrderType orderType;
    private BigDecimal limitPrice;
    private PositionSide positionSide;
    private List<StopOrder> takeProfits;
    private List<StopOrder> stopLosses;

    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class StopOrder {
        private String orderId;
        private BigDecimal quantity;
        private BigDecimal price;
    }
}

File: dxtrade-extension-frontend/src/dto/internal/request/PlaceOrderRequest.ts
import { Broker } from "../Broker";

export type PlaceOrderRequest = {
    // broker: Broker,
    broker: string,
    apiKey: string;
    orderId: string,
    accountCode: string;
    quantity: number;
    orderType: 'LIMIT' | 'MARKET';
    limitPrice?: number | string; 
    instrument: string;
    positionSide: PositionSide;
    takeProfits: StopOrder[];
    stopLosses: StopOrder[];
}

export type ModifyOrderRequest = {
    broker: string,
    apiKey: string;
    orderCode: string,
    accountCode: string;
    quantity: number;
    orderSide: 'BUY' | 'SELL';
    limitPrice?: number | string; 
    instrument: string,
    orderType: string,
    positionCode: string
}

export type StopOrder = {
    orderId: string;
    quantity: number;
    price: number | string;
}

export enum PositionSide {
    LONG,
    SHORT
}